sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -631,6 +674,8 @@ kernel_pid_t gcoap_init(void)
     /* randomize initial value */
     atomic_init(&_coap_state.next_message_id, (unsigned)random_uint32());
 
+    nanocoap_cache_init();
```

Comment:
Besides having the `NOP` functions when the module is not used, maybe to improve readability you can enclose the calls with:
```C
if (IS_ACTIVE(MODULE_NANOCOAP_CACHE)) {
    nanocoap_cache_init();
}
```

Reply:
Makes sense. I added a C-guard around `nanocoap_cache_init()` for now. The integration into gcoap is still experimental. Once we figure out a good way (including the forward proxy), then I will apply the guarding also there. Thanks!